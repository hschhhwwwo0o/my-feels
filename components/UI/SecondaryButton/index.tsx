import React, { FunctionComponent, ReactNode } from "react";

interface ISecondaryButton {
  onClickHandler?: () => any;
  children?: ReactNode;
}

const SecondaryButton: FunctionComponent<ISecondaryButton> = ({
  children,
  onClickHandler = () => {},
}) => {
  return (
    <div>
      <button
        onClick={onClickHandler}
        className="w-full bg-[#FAFAFA] text-[#2F2F2F] py-[21px] rounded-md text-base px-5 font-semibold border-2 border-[#EAEAEA] dark:border-[#151515] transition-all duration-1000"
      >
        {children}
      </button>
    </div>
  );
};

export default SecondaryButton;
