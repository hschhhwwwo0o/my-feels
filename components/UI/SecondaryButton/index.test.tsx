import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import SecondaryButton from "./index";

test("Test SecondaryButton component", () => {
  render(<SecondaryButton onClickHandler={() => {}}>Secondary text</SecondaryButton>);
  const el: HTMLElement = screen.getByRole("button");
  expect(el).toBeInTheDocument();
  expect(el).toMatchSnapshot();
});
