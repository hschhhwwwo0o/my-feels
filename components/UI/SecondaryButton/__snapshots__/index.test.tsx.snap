// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SecondaryButton component 1`] = `
<button
  class="w-full bg-[#FAFAFA] text-[#2F2F2F] py-[21px] rounded-md text-base px-5 font-semibold border-2 border-[#EAEAEA] dark:border-[#151515] transition-all duration-1000"
>
  Secondary text
</button>
`;
